core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -189,6 +189,9 @@ UNWIND(	.cantunwind)
 	orr	r0, r0, #SCTLR_A
 	bic	r0, r0, #SCTLR_C
 	bic	r0, r0, #SCTLR_I
+#if defined(CFG_HWSUPP_MEM_PERM_WXN) && defined(CFG_CORE_RWDATA_NOEXEC)
```

Comment:
Better to test only `CFG_CORE_RWDATA_NOEXEC` here.
Optionally, add a dependency check in a `core/arc/arm/arm.mk` (but see my other comment, I feel the HWSUPP config is useless):
```
ifeq ($(CFG_CORE_RWDATA_NOEXEC))
ifneq ($(CFG_HWSUPP_MEM_PERM_WXN),y)
$(error ...)
endif
endif
```